packages/zowe-explorer-api/src/fs/types/datasets.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { DirEntry, EntryMetadata, FileEntry } from "../types";
+import { IProfileLoaded } from "@zowe/imperative";
+import { FileType } from "vscode";
+
+export class DsEntry extends FileEntry {
+    public name: string;
+    public metadata: DsEntryMetadata;
+    public type: FileType;
+    public wasAccessed: boolean;
```

Comment:
I may be missing something very important, but do we need to re-difine every property defined in the extended class?
I get the metadata is being overloaded, but most/all other properties seem exactly the same 😋